src/lib/types.ts

```diff
@@ -185,19 +185,21 @@ export interface UserCredentials {
   oidc?: OpenIDConnectCredentials
 }
 
-export interface VerifyOTPParams {
-  email?: string
-  phone?: string
+export type VerifyOTPParams = VerifyMobileOTPParams | VerifyEmailOTPParams
+export interface VerifyMobileOTPParams {
+  email?: undefined
+  phone: string
+  token: string
+  type?: MobileOTPType
+}
+export interface VerifyEmailOTPParams {
+  email: string
+  phone?: undefined
   token: string
-  type?: OTPType
+  type: EmailOTPType
 }
-export type OTPType = 
-  | 'signup' 
-  | 'invite' 
-  | 'magiclink' 
-  | 'recover' 
-  | 'email_change' 
-  | 'sms'
+export type MobileOTPType = 'sms'
+export type EmailOTPType = 'signup' | 'invite' | 'magiclink' | 'recover' | 'email_change'
```

Comment:
@kangmingtay can you double check that this separation is correct?

Reply:
hmm seems like it should be 'recovery' instead of 'recover' (https://github.com/supabase/gotrue/blob/master/api/verify.go#L24)